Userland/Libraries/LibGUI/FileSystemModel.cpp

```diff
@@ -700,8 +701,18 @@ static Threading::MutexProtected<ThumbnailCache> s_thumbnail_cache {};
 
 static ErrorOr<NonnullRefPtr<Gfx::Bitmap>> render_thumbnail(StringView path)
 {
-    Gfx::IntSize thumbnail_size { 32, 32 };
-    auto bitmap = TRY(Gfx::Bitmap::load_from_file(path, 1, thumbnail_size));
+    Core::EventLoop event_loop;
+    Gfx::IntSize const thumbnail_size { 32, 32 };
+
+    auto file = TRY(Core::MappedFile::map(path));
+    auto client = TRY(ImageDecoderClient::Client::try_create());
```

Comment:
Hmm wait, does this make a new IPC connection for every thumbnail? Can we cache and reuse an existing one (as long as it doesn't crash ImageDecoder ofc)?

Reply:
That's indeed nicer!